src/provisioner/cluster.py

```diff
@@ -201,7 +213,8 @@ def save(self, cortx_conf):
                 nodes = storage_set['nodes']
                 for node_idx, node in enumerate(nodes):
                     # confstore keys
-                    kvs.append((f'{key_prefix}>nodes[{node_idx}]', node['id']))
+                    node_id = CortxCluster.get_machine_id(node['id'])
+                    kvs.append((f'{key_prefix}>nodes[{node_idx}]', node_id))
```

Comment:
I think machine_id needs to reflect in GConf and not to the solution supplied config. 
Key would be "storage_sets[i]>nodes[j]>id".

Reply:
done, I have replcaed it with machine_id now.